print-ptp.c

```diff
@@ -0,0 +1,619 @@
+ *    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+                               +
+ *    |      Response Origin Time Stamp Seconds                       |
+ *    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ *    |                         Nano Seconds                          |
+ *    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ *    | Requesting Port Identity      |
+ *    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ *
+ *  Signalling Message (msg type=0xC)
+ *     0                   1                   2                   3
+ *     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+ *                                    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ *                                    | Requesting Port Identity      |
+ *    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ *
+ *  Management Message (msg_type=0xD)
+ *     0                   1                   2                   3
+ *     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+ *                                    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ *                                    | Requesting Port Identity      |
+ *    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ *    |Start Bndry Hps| Boundary Hops | flags         | Reserved      |
+ *    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ *
+ */
+
+#define M_SYNC                  0x0
+#define M_DELAY_REQ             0x1
+#define M_PDELAY_REQ            0x2
+#define M_PDELAY_RESP           0x3
+#define M_OTHER                 0x5
+#define M_FOLLOW_UP             0x8
+#define M_DELAY_RESP            0x9
+#define M_PDELAY_RESP_FOLLOW_UP 0xA
+#define M_ANNOUNCE              0xB
+#define M_SIGNALLING            0xC
+#define M_MANAGEMENT            0xD
+
+static const struct tok ptp_msg_type[] = {
+    { M_SYNC ,"sync msg"},
+    { M_DELAY_REQ ,"delay req msg"},
+    { M_PDELAY_REQ ,"peer delay req msg"},
+    { M_PDELAY_RESP ,"peer delay resp msg"},
+    { M_OTHER, "Other"},
+    { M_FOLLOW_UP ,"follow up msg"},
+    { M_DELAY_RESP ,"delay resp msg"},
+    { M_PDELAY_RESP_FOLLOW_UP ,"pdelay resp fup msg"},
+    { M_ANNOUNCE ,"announce msg"},
+    { M_SIGNALLING ,"signalling msg"},
+    { M_MANAGEMENT ,"management msg"},
+    { 0, NULL}
+};
+
+
+#define PTP_TRUE 1
+#define PTP_FALSE !PTP_TRUE
+
+#define PTP_HDR_LEN         0x22
+
+/* mask based on the first byte */
+#define PTP_VERS_MASK       0xFF
+#define PTP_V1_COMPAT       0x10
+#define PTP_MSG_TYPE_MASK   0x0F
+
+/*mask based 2byte */
+#define PTP_DOMAIN_MASK     0xFF00
+#define PTP_RSVD1_MASK      0xFF
+#define PTP_CONTROL_MASK    0xFF
+#define PTP_LOGMSG_MASK     0xFF
+
+/* mask based on the flags 2 bytes */
+
+#define PTP_L161_MASK               0x1
+#define PTP_L1_59_MASK              0x2
+#define PTP_UTC_REASONABLE_MASK     0x4
+#define PTP_TIMESCALE_MASK          0x8
+#define PTP_TIME_TRACABLE_MASK      0x10
+#define PTP_FREQUENCY_TRACABLE_MASK 0x20
```

Comment:
Looking at the PTP description: https://support.huawei.com/hedex/pages/EDOC100010596830008125/05/EDOC100010596830008125/05/resources/message/cd_feature_1588v2_format-general.html#cd_feature_eth_5__packetformat

Was this meant to be described as `PTP_TIME_TRACEABLE_MASK` and `PTP_FREQUENCY_TRACEABLE_MASK`

Reply:
> Please, use the new GET_ macros instead of EXTRACT_ ones

Use the GET_  macros, not the get_ inline functions.